File: src/main/java/com/github/krystianmuchla/home/pagination/api/PaginationRequest.java
package com.github.krystianmuchla.home.pagination.api;

import com.github.krystianmuchla.home.pagination.Pagination;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public final class PaginationRequest {
    @NotNull @Size(min = Pagination.MIN_PAGE_NUMBER) Integer pageNumber = Pagination.DEFAULT_PAGE_NUMBER;
    @NotNull @Size(min = Pagination.MIN_PAGE_SIZE, max = Pagination.MAX_PAGE_SIZE) Integer pageSize = Pagination.DEFAULT_PAGE_SIZE;
}

File: src/main/kotlin/com/example/skyr/pagination/api/PaginationRequest.kt
package com.example.skyr.pagination.api

import com.example.skyr.pagination.Pagination
import jakarta.validation.constraints.Size

data class PaginationRequest(
    @field:Size(min = Pagination.MIN_PAGE_NUMBER) val pageNumber: Int = Pagination.DEFAULT_PAGE_NUMBER,
    @field:Size(
        min = Pagination.MIN_PAGE_SIZE,
        max = Pagination.MAX_PAGE_SIZE
    ) val pageSize: Int = Pagination.DEFAULT_PAGE_SIZE
)
